18 create automated test cases

KMarshall33 opened the pull request:
Hi @nnilf & @dombrown95 - Please could I have your review on the changes to the automated test cases linked to #41. I have changed the following:

- Created a shared TestBase class to centralise common setup/teardown logic such as creating and cleaning temporary directories.
- Updated all test files (Test_Alert_Module.py, Test_Database.py, Test_Sensor_Integration.py, Test_Fault_Detection.py) to inherit from TestBase instead of unittest.TestCase.
- Removed duplicated setUp() / tearDown() code across test files.
- Applied polymorphism by overriding setUp() or tearDown() in modules that required additional behaviour (e.g. closing database connections) while still calling super().

KMarshall33 committed:
Add base test file

KMarshall33 committed:
Merge branch 'main' of https://github.com/TriGuys/aerospace-sim-engine into 18-create-automated-test-cases

KMarshall33 committed:
Inherit from Test_Base in Test_Alert_Module.py.

KMarshall33 committed:
Inherit from Test_Base in Test_Sensor_Integration.py.

KMarshall33 committed:
Inherit from Test_Base in Test_Fault_Detection.py.

KMarshall33 committed:
Inherit from Test_Base in Test_Database.py.

dombrown95 reviewed (changes requested):
Hi @KMarshall33 great work on implementing the base class - the code so far looks really well organised and clean. I have checked our current test case coverage and it looks like we'll have to add some more tests to ensure we can properly perform requirements based testing. 

As it stands, we have the following requirements covered: 

- FR1
- FR2
- FR3
- FR5
- NFR4

Not covered so far:
- FR4: Display alerts raised in GUI 
- FR6 : Mark alerts as resolved/unresolved
- FR7: Sort alerts by severity in GUI
- FR8: Fast upload and alert display (could add performance/time based test)
- NFR1: Consistency between GUI and DB
- NFR3: Logging of operations and errors
- NFR5: Load and process flight data within 5 seconds
- NFR2: This one can be tested manually as it involves the GUI being clearly structured/easy to navigate.

Please could you add test cases for the above - thanks!

nnilf commented on Tests/Test_Fault_Detection.py:
```diff
@@ -2,30 +2,22 @@
-        self.tmpdir = tempfile.TemporaryDirectory()
-        self.tmp_path = Path(self.tmpdir.name)
+        super().setUp()
         self.fd = FaultDetection()
```
To stick with coding standards, please provide a more descriptive name e.g. fault_detection.

KMarshall33 replied:
Changed fd to fault_detection.

nnilf commented on .vs/slnx.sqlite:
Add file to gitignore.

KMarshall33 replied:
Added to gitignore.

nnilf commented on Tests/Test_Alert_Module.py:
```diff
@@ -2,26 +2,24 @@
 
-class TestAlertModule(unittest.TestCase):
+class TestAlertModule(TestBase):
     def setUp(self):
```
Additionally function names need to be snake_case, as outlined by coding standards https://github.com/TriGuys/aerospace-sim-engine/issues/19

KMarshall33 replied:
Added type hinting. setUp is from unittest therefore, it must be camelCase.

nnilf commented on Tests/Test_Alert_Module.py:
```diff
@@ -2,26 +2,24 @@
         self.db = AlertDatabase(str(self.db_path))
         self.mod = AlertModule(self.db)
 
     def tearDown(self):
```
Additionally function names need to be snake_case, as outlined by coding standards https://github.com/TriGuys/aerospace-sim-engine/issues/19

nnilf commented on Tests/Test_Alert_Module.py:
Please add type hinting to meet coding standards, def set_up(self) -> None:

present throughout all the tests.

nnilf commented on Tests/Test_Base.py:
Please add type hinting to meet coding standards, def set_up(self) -> None:

present throughout all the tests

nnilf commented on Tests/Test_Base.py:
function names need to be snake_case, as outlined by coding standards https://github.com/TriGuys/aerospace-sim-engine/issues/19

KMarshall33 replied:
setUp and tearDown are from unittest therefore, they must be camelCase.

nnilf commented on Tests/Test_Database.py:
uses a magic number, need to assign value to a descriptive variable name prior, which should also remove the need for the comment.

nnilf commented on Tests/Test_Database.py:
Line 74 says descending when this has been changed previously.

nnilf commented on Tests/Test_Fault_Detection.py:
```diff
@@ -2,30 +2,22 @@
     def test_rules_loaded(self):
         """Test that fault detection rules are loaded correctly."""
-        self.assertEqual(len(self.fd.detectionRules), 7)
+        self.assertEqual(len(self.fd.detection_rules), 7)
```
This just tests 7 exist, not that they have been loaded correctly.

KMarshall33 replied:
Changed test to verify that rules load with the required keys.

nnilf commented on Tests/Test_Sensor_Integration.py:
Please add type hinting to meet coding standards, def set_up(self) -> None:

present throughout all the tests

nnilf commented on Tests/Test_Sensor_Integration.py:
function names need to be snake_case, as outlined by coding standards https://github.com/TriGuys/aerospace-sim-engine/issues/19

nnilf reviewed (changes requested):
Changes requested

KMarshall33 committed:
Add base User Interface test file.

KMarshall33 committed:
Add files to gitignore.

KMarshall33 commented on Tests/Test_Sensor_Integration.py:
```diff
@@ -3,33 +3,16 @@
-        self.tmpdir = tempfile.TemporaryDirectory()
-        self.tmp_path = Path(self.tmpdir.name)
+        super().setUp()
         self.si = SensorIntegration()
```
Changed si to sensor_integration.

KMarshall33 committed:
Add type hinting and clearer variable names to Test_Base.py.

KMarshall33 committed:
Add type hinting and clearer variable names to Test_Alert_Module.py.

KMarshall33 commented on Tests/Test_Database.py:
Added type hinting to all tests.

KMarshall33 commented on Tests/Test_Database.py:
setUp and tearDown are from unittest therefore, they must stay in camelCase.

KMarshall33 committed:
Add type hinting and clearer variable names to Test_Database.py.

KMarshall33 committed:
Assign magic numbers to descriptive variables in Test_Alert_Module.py and Test_Database.py.

KMarshall33 commented on Tests/Test_Alert_Module.py:
Assigned magic number to a descriptive variable and removed comment.

KMarshall33 committed:
Add type hinting and clearer variable names in Test_Fault_Detection.py.

KMarshall33 commented on Tests/Test_Fault_Detection.py:
Added type hinting to all tests.

KMarshall33 committed:
Add type hinting and clearer variable names to Test_Sensor_Integration.py.

KMarshall33 committed:
Add test for NFR4 and improve FR1 test coverage in Test_Fault_Detection.py.

KMarshall33 committed:
Add comments for requirement traceability for Test_Database.py.

KMarshall33 committed:
Add a check for logging error in Test_Database.py.

KMarshall33 committed:
Add and improve tests in Test_Sensor_Integration.py for better requirements coverage.

KMarshall33 committed:
Correct logging.warning to logging.error to satisfy testing requirement for SensorIntegration.py.

KMarshall33 committed:
Change Test_Fault_Detection.py to check for object attributes rather than dictionary keys.

KMarshall33 committed:
Add tests for better coverage on FR4 & FR5, improve current tests in Test_Alert_Module.py.

KMarshall33 committed:
Merge branch 'main' of https://github.com/TriGuys/aerospace-sim-engine into 18-create-automated-test-cases

KMarshall33 committed:
Change test status check to use database status rather than AlertModule cache status and ensure status in database is updated upon change.

dombrown95 committed:
add initial setUp and tearDown methods for testing UserInterface

dombrown95 committed:
update UserInterface test cases to test for critical and moderate alerts

dombrown95 committed:
add test to confirm deleting alert removes from both database and the GUI

dombrown95 committed:
update test case to prevent GUI pop up when running

dombrown95 committed:
add test to confirm resolve_alert marks alerts as resolved in database

dombrown95 committed:
add test case to confirm resolve_alert switches an already resolved alert back to active

dombrown95 committed:
add test case to confirm alert table matches database alerts after modification

dombrown95 committed:
add test to confirm UserInterface module can be instantiated independently (NFR4)

dombrown95 committed:
add SensorIntegration test case to confirm CSV data is read/cleaned within 5 seconds (FR8, NFR5)

KMarshall33 committed:
Remove logging check for invalid timestamp test in Test_Database.py (not required).

KMarshall33 commented:
Hi @nnilf & @dombrown95 - I have addressed all your comments. Thanks @dombrown95 for taking the UI test file.

All of the tests are working - please could you both check the coverage of the requirements and confirm you're happy with these tests so we can start testing the modules.

Thanks.

nnilf committed:
Merge branch 'main' of github.com-home:TriGuys/aerospace-sim-engine into 18-create-automated-test-cases
